### ISSUE TITLE
S3 DeleteObjects failing using Go AWS SDK V2

### ISSUE THREAD

**Issue by shawnHartsell** [localstack/localstack#4015]
# Type of request: This is a ...

[ x ] bug report
[ ] feature request

# Detailed description

I am porting a service from `github.com/aws/go-aws-sdk` to `github.com/aws/go-aws-sdk-v2` and encountering failures when making `DeleteObjects` calls to S3. I have validated that making a similar call through the AWS CLI does not fail

```
aws --endpoint-url http://localhost:4566 s3api delete-objects --bucket my-bucket --delete '{"Objects":[{"Key": "key-1"},{"Key": "key-2"}]}'

```

## Expected behavior

`RemoveObjects` calls to S3 using the Go AWS SDK V2 do not fail

## Actual behavior

Below are the request/response logs from the the V2 Go SDK as well as the stack trace from my `localstack` container:

**Request via Go SDK V2**

```
POST /my-bucket?delete=&x-id=DeleteObjects HTTP/1.1
Host: localhost:4566
User-Agent: aws-sdk-go-v2/1.3.2
Content-Length: 132
Accept-Encoding: identity
Amz-Sdk-Invocation-Id: b1884ab5-4df0-40cd-bf78-506b739425df
Amz-Sdk-Request: attempt=3; max=3
Authorization: AWS4-HMAC-SHA256 Credential=cat/20210512/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;amz-sdk-invocation-id;amz-sdk-request;content-length;content-md5;content-type;host;x-amz-content-sha256;x-amz-date;x-amz-user-agent, Signature=93b13eaa0628119eaebd2851d461c2016fe8d85b277c0f5db4a027cc470c8d9c
Content-Md5: HQxJAOODzcZOBxnrKCy+7A==
Content-Type: application/xml
X-Amz-Content-Sha256: 9886c84d28e2c216cf9c4027ee4520f0e64d9ceafd8c6bc745ed544cc7ddbf40
X-Amz-Date: 20210512T232646Z
X-Amz-User-Agent: aws-sdk-go-v2/1.3.2 os/other lang/go/1.16.3 md/GOOS/darwin md/GOARCH/amd64

<Delete xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Object><Key>f0/f02ed11b-991d-4c66-a9c8-b22001f8aa1c</Key></Object></Delete>
```

**Response via Go SDK V2 from localstack**

```
HTTP/1.1 500 
Connection: close
Content-Length: 287
Access-Control-Allow-Headers: authorization,content-type,content-length,content-md5,cache-control,x-amz-content-sha256,x-amz-date,x-amz-security-token,x-amz-user-agent,x-amz-target,x-amz-acl,x-amz-version-id,x-localstack-target,x-amz-tagging
Access-Control-Allow-Methods: HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: x-amz-version-id
Content-Type: text/html; charset=utf-8
Date: Wed, 12 May 2021 23:26:46 GMT
Last-Modified: Wed, 12 May 2021 23:26:46 GMT
Server: hypercorn-h11
X-Amz-Id-2: MzRISOwyjmnup02192EAC2F7533C27/JypPGXLh0OVFGcJaaO3KW/hRAqKOpIEEp
X-Amz-Request-Id: 02192EAC2F7533C2

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN"><title>500 Internal Server Error</title><h1>Internal Server Error</h1><p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>
```

**Stack Trace From localstack**

```
2021-05-12 23:26:46,083:API: Error on request:
Traceback (most recent call last):
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 323, in run_wsgi
    execute(self.server.app)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/serving.py", line 312, in execute
    application_iter = app(environ, start_response)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/server.py", line 177, in __call__
    return backend_app(environ, start_response)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/core/utils.py", line 156, in __call__
    result = self.callback(request, request.url, {})
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/s3/responses.py", line 256, in ambiguous_response
    return self.bucket_response(request, full_url, headers)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/s3/responses.py", line 265, in bucket_response
    response = self._bucket_response(request, full_url, headers)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/s3/responses.py", line 317, in _bucket_response
    return self._bucket_response_post(request, body, bucket_name)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/moto/s3/responses.py", line 869, in _bucket_response_post
    key = form["key"]
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/werkzeug/datastructures.py", line 442, in __getitem__
    raise exceptions.BadRequestKeyError(key)
werkzeug.exceptions.BadRequestKeyError: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.
KeyError: 'key'
```

# Steps to reproduce

## Command used to start LocalStack

Below is a snippet from our `docker-compose.yml`

```
localstack:
    container_name: localstack
    image: localstack/localstack:0.12.11
    env_file:
      - ./env/localstack.env
    ports:
      - "4566:4566"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
```

Below is our `.env`

```
SERVICES=s3
DEFAULT_REGION=us-east-1
LAMBDA_EXECUTOR=docker
DEBUG=1
AWS_ACCESS_KEY_ID=local
AWS_SECRET_ACCESS_KEY=local
AWS_DEFAULT_REGION=us-east-1
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

Here's an abbreviated snippet on how we are setting up our S3 client using the Go AWS V2 SDK with localstack:

```
	awsCfg, err := awsconfig.LoadDefaultConfig(context.TODO(),
		awsconfig.WithClientLogMode(aws.LogRequestWithBody | aws.LogResponseWithBody),
		awsconfig.WithRegion("us-east-1"),
		awsconfig.WithEndpointResolver(aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
			return aws.Endpoint{
				PartitionID:   "aws",
				URL:           "http://localhost:4566,
				SigningRegion: "us-east-1",
			}, nil
		})))

	if err != nil {
       panic(err)
    }

	client := s3.NewFromConfig(awsCfg,
		func(options *s3.Options) {
			options.UsePathStyle = true
			options.EndpointOptions.DisableHTTPS = true
		})


    input := s3.DeleteObjectsInput{
		Bucket: aws.String("my-bucket"),
		Delete: &s3types.Delete{
			Objects: []types.ObjectIdentifier{
			{ Key: aws.String("key-1") },
			{ Key: aws.String("key-2") },
		},
	  },
	}

   output, err := s.AWSClient.DeleteObjects(context.Background(), &input)
   if err != nil {
       panic(err)
   }

   fmt.Printf("%+v\n", *output)

```

**Comment by shawnHartsell** [comment]
Looks like the issue could be related to https://github.com/localstack/localstack/issues/3931

**Comment by macnev2013** [comment]
Thanks for reporting @shawnHartsell . This should be fixed in #4140 - can you please do a `docker pull localstack/localstack` and give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
